operations/research-and-development/product/user-interface-text-guidelines.md

```diff
@@ -0,0 +1,266 @@
+# User Interface Text Guidelines
+
+## UX Writing Principles
+
+Following the Writing Guidelines below will ensure communications to our users feel like they are coming from the same voice.
+
+**Concise**
+
+Use short words and sentences. Avoid unnecessary fluff.
+
+**Clear**
+
+Always write for clarity and simplicity. Err on the side of simple language that can be understood by novice users and avoid vague terms.
+
+**Consistent**
+
+Be consistent with terminology across the product. Although we may have different terms for the same things internally, always use the same language to describe the same thing within the product.
+
+## User interface text guidelines 
+
+These guidelines cover two types of UI text:
+- Messages triggered because something happened
+- Static text on the user interface
+
+### General principles for UI (in-product) text
+
+- Keep in mind that your users are from all over the world. 
+- Use the present tense to describe a current state or condition, and the future tense to state something that is very definitely going to happen.
+- Use the active voice, except for these cases:
+  - If you’ll end up blaming the user. For example, don’t say **You entered an incorrect password**. Instead, say **The password is incorrect**.
+  - If you’re describing what just happened. For example, it’s okay to say **Your incoming webhook is set up**.
+  - If the subject (the doer of an action) is the Mattermost application itself. For example, it’s okay to say **The image is deleted** instead of **The server deleted the image**.
+- If you’re asking the user to do or not do something, use imperatives (command phrases). For example, say **Do not change the Hostname** instead of saying **It is not recommended to change the Hostname**. Better still, explain what could go wrong if they do or don’t do something. For example, say **Do not change the Hostname because if you do so <this bad thing will happen>**.
```

Comment:
```suggestion
- If you’re asking the user to do or not do something, use imperatives (command phrases). For example, use **Don't change the Hostname** instead of **It's not recommended to change the Hostname**. Better still, explain what could go wrong if they do or don’t do something. For example: **Don't change the Hostname because doing so <could cause this bad thing will happen>**.
```